client-js/app/elements/labrad-registry.ts

```diff
@@ -380,22 +374,39 @@ export class LabradRegistry extends polymer.Base {
     }
   }
 
-  async doDragCopy() {
-    var newName =  this.$.dragCopyNameInput.value;
-    var newPath = JSON.parse(this.$.dragCopyPathInput.value);
+  /**
+   * Execute the drag operation
+   */
+  async doDragOp() {
+    var newName =  this.$.dragNameInput.value;
+    var newPath = JSON.parse(this.$.dragPathInput.value);
     var oldPath = JSON.parse(this.$.originPath.textContent);
     var oldName = this.$.originName.textContent;
 
-    this.$.pendingDialog.open();
-    this.$.pendingOp.innerText = "Copying...";
-
-    try {
-      var resp = await this.socket.copyDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
-      console.log(resp)//this.repopulateList();
-      this.$.pendingDialog.close()
-      this.$.toastTextSuccess.show();
-    } catch (error) {
-      this.handleError(error);
+    if (this.$.dragOp.innerText === 'Copy') {
+      try {
+        this.$.pendingDialog.open();
+        this.$.pendingOp.innerText = "Copying...";
+        console.log(this.$.dragOp.innerText)
+        var resp = await this.socket.copyDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
+        console.log(resp)
+        this.$.pendingDialog.close()
```

Comment:
I would put this close call in a finally block so it happens whether the copy succeeds or fails.